### PR TITLE
Start slash command permission

### PR DESCRIPTION
**Pull request by d-nery**
This starts slash command permission management

Permissions only applies in guild commands, so I modified the guild_create_command to also send the guild_command_edit_permission after it has completed succesfully

Relevant docs: https://discord.com/developers/docs/interactions/slash-commands#get-guild-application-command-permissions

### CODE REVIEW

File: include/dpp/slashcommand.h [hunk 1]
```diff
@@ -343,12 +343,67 @@ class interaction : public managed {
...
 /**
  * @brief Represents an application command, created by your bot
  * either globally, or on a guild.
  */
 class slashcommand : public managed {
 public:
+	/**
+	 * @brief Command id (returned from discord api upon registering)
+	 */
+	snowflake id;
```
Comment (braindigitalis) [inline include/dpp/slashcommand.h:405]: not needed, managed::id already defines this field

**Review by braindigitalis — CHANGES_REQUESTED** [review]
looks good except for duplicated id member variable in slashcommand class

**Comment by webtax-gh** [comment]
@d-nery is this still a draft 🚧, or is it ready 😄! Code looks good and it's ready to merge if you want to.

**Comment by d-nery** [comment]
I'll do some final checks, get the lasat changes from dev and solve conflicts, and then I remove draft

**Comment by braindigitalis** [comment]
is this ready to merge in now? seemed a simple commit

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**